website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,158 @@
+
+### Preparation for upgrade
+
+Upgrade preparation consists of several actions, and is performed using [nsxt_upgrade_prepare](../resources/upgrade_prepare.html.markdown):
+* Uploading the upgrade and precheck bundles.
+* Acceptance of the user agreement.
+* Reporting of failed pre-checks.
+
+```hcl
+resource "nsxt_upgrade_prepare" "prepare_res" {
+  upgrade_bundle_url    = "http://url-to-upgrade-bundle.mub"
+  precheck_bundle_url   = "http://url-to-precheck-bundle.pub"
+  accept_user_agreement = true
+}
+```
+
+**NOTE:** Acceptance of the license agreement is required to initiate the upgrade process.
+**NOTE:** Precheck bundle is usable only while upgrading a NSX manager of version v4.1.1 and above.
+
+### Acknowledgement of upgrade issues and warnings
+
+The preparation stage may result with various warnings which would halt the upgrade process. These warnings can be 
+suppressed with the `nsxt_upgrade_precheck_acknowledge` resource.
+
+For example:
+```
+There are unacknowledged warnings in precheck: [backupOperationCheck], please address these errors from NSX or using nsxt_upgrade_precheck_acknowledge resource
+```
+
+The resource uses the `precheck_ids` to conclude which prechecks are safe to ignore for this upgrade cycle.
+The `nsxt_upgrade_precheck_acknowledge` resource requires the `target_version` from the `nsxt_upgrade_prepare` resource 
+to make sure that the acknowledged warnings are from the correct execution, and to enforce execution of the 
+`nsxt_upgrade_precheck_acknowledge` resource after preparation is complete.
+
+The `nsxt_upgrade_prepare_ready` data source verifies that the preparation phase of the upgrade was concluded 
+successfully and enable the execution of the upgrade. 
+
+```hcl
+resource "nsxt_upgrade_precheck_acknowledge" "test" {
+  precheck_ids   = ["backupOperationCheck", "pUBCheck"]
+  target_version = nsxt_upgrade_prepare.test.target_version
+}
+
+data "nsxt_upgrade_prepare_ready" "ready" {
+  upgrade_prepare_id = nsxt_upgrade_prepare.prepare_res.id
+  depends_on         = [nsxt_upgrade_precheck_acknowledge.test]
+}
+```
+
+### Running the upgrade
+
+In order to configure and execute upgrade of NSXT edges, hosts, and managers, use `nsxt_upgrade_run` resource. 
+
+This resource will start with configuration of upgrade unit groups and upgrade plan settings for `EDGE`and `HOST` 
+components. Host component groups may be created within this resource with custom host lists and attributes, or it can 
+use the predefined groups which NSX provides.
+
+Following the configuration of the edge, host groups, it will execute the upgrade for each component. If there are 
+either `EDGE` or `HOST` groups disabled from the upgrade, the corresponding component will be paused after enabled 
+groups get upgraded. At the same time, `MP` upgrade won't be processed, because that requires `EDGE` and `HOST` to be 
+fully upgraded. Refer to the exposed attribute for current upgrade state details. For example, check `upgrade_plan`
```

Comment:
We can also have multiple nsxt_ugprade_run resources? In that case could we add to the sample below a 2nd resource that will upgrade the disabled edge group and then MP?

Reply:
I don't think we should support multiple `upgrade` or `prepare` resource instances, this would complicate the logic